tests/ui/lint/static-mut-refs.e2021.stderr

```diff
@@ -1,4 +1,4 @@
-warning: creating a shared reference to mutable static is discouraged
+warning: creating a shared reference to mutable static is not allowed
```

Comment:
I would like to ask if we should keep the error message as "discouraged" since my update will change the existing message in 2021 Edition 🤔 This is the first time to contribute the diagnostic issue, it would be nice if someone could tell me about the compatibility thing.

Reply:
The easy way is to reword this message to...

```
warning: creating a shared reference to mutable static
```

that way, it doesn't really matter if it's Edition 2021 / 2024.

The more complex way is to introduce different diagnostics between edition < 2024 and edition >= 2024, with edition 2021 wording being

```
warning: creating a shared reference to mutable static is discouraged
```

and edition 2024 wording being

```
error: creating a shared reference to mutable static is not allowed
```

Reply:
I agree that `creating a shared reference to mutable static` is better.

The `warning: ` and `error: ` will be enough information to let people know whether it's discouraged or not allowed.

("error" implies not being allowed and "warning" implies being discouraged)